compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1026,6 +1066,44 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             "the {item_kind} was found for\n{}{}",
                             type_candidates, additional_types
                         ));
+                    } else {
+                        let mut candidate_confusable = None;
+
+                        for inherent_impl_did in self.tcx.inherent_impls(adt.did()) {
+                            for inherent_method in
+                                self.tcx.associated_items(inherent_impl_did).in_definition_order()
+                            {
+                                if let Some(Attribute { kind: AttrKind::Normal(attr), .. }) = self
+                                    .tcx
+                                    .get_attr(inherent_method.def_id, sym::rustc_confusables)
+                                {
+                                    match ConfusablesDirective::try_parse(attr) {
+                                        Ok(c) => {
+                                            if c.names.contains(&item_name.to_string()) {
+                                                candidate_confusable = Some(inherent_method);
+                                                break;
+                                            }
+                                        }
+                                        Err(e) => {
+                                            self.sess().emit_err(e);
```

Comment:
This is not the right place to emit this error. It may be related to an attribute in a different crate.
Also, we don't report malformed attributes in the current crate if they are not used.
Such reporting should go in `rustc_passes::check_attr`.